clmm/support/mock_data.py

```diff
@@ -134,20 +136,20 @@ def generate_galaxy_catalog(cluster_m, cluster_z, cluster_c, cosmo, zsrc, Delta_
 
     galaxy_catalog = _generate_galaxy_catalog(ngals=ngals, **params)
     # Check for bad galaxies and replace them
+    nbad, badids = _find_aphysical_galaxies(galaxy_catalog, zsrc_min)
     for i in range(nretry):
-        nbad, badids = _find_aphysical_galaxies(galaxy_catalog, zsrc_min)
         if nbad < 1:
             break
         replacements = _generate_galaxy_catalog(ngals=nbad, **params)
         #galaxy_catalog[badids] = replacements
         for badid, replacement in zip(badids, replacements):
             for col in galaxy_catalog.colnames:
                 galaxy_catalog[col][badid] = replacement[col]
+        nbad, badids = _find_aphysical_galaxies(galaxy_catalog, zsrc_min)
 
     # Final check to see if there are bad galaxies left
-    nbad, _ = _find_aphysical_galaxies(galaxy_catalog, zsrc_min)
     if nbad > 1:
-        print("Not able to remove {} aphysical objects after {} iterations".format(nbad, nretry))
+        warnings.warn("X Not able to remove {} aphysical objects after {} iterations".format(nbad, nretry))
```

Comment:
Typo in the warning message: need to remove 'X'

Reply:
Good catch. Fixed.